client/my-sites/comments/comment-list/comment-list-header.jsx

```diff
@@ -0,0 +1,95 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import HeaderCake from 'components/header-cake';
+import QueryPosts from 'components/data/query-posts';
+import { convertDateToUserLocation } from 'components/post-schedule/utils';
+import { decodeEntities, stripHTML } from 'lib/formatting';
+import { gmtOffset, timezone } from 'lib/site/utils';
+import { bumpStat, composeAnalytics, recordTracksEvent } from 'state/analytics/actions';
+import { getSiteComments } from 'state/selectors';
+import { getSitePost } from 'state/posts/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { getSelectedSite, getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+
+export const CommentListHeader = ( {
+	postDate,
+	postId,
+	postTitle,
+	postUrl,
+	recordReaderArticleOpened,
+	site,
+	siteId,
+	siteSlug,
+	translate,
+} ) => {
+	const formattedDate = postDate
+		? convertDateToUserLocation( postDate, timezone( site ), gmtOffset( site ) ).format( 'll LT' )
+		: '';
+
+	return (
+		<div className="comment-list__header">
+			<QueryPosts siteId={ siteId } postId={ postId } />
+
+			<HeaderCake
+				actionHref={ postUrl }
+				actionIcon="visible"
+				actionOnClick={ recordReaderArticleOpened }
+				actionText={ translate( 'View Post' ) }
+				backHref={ `/comments/all/${ siteSlug }` }
+			>
+				<div className="comment-list__header-title">{ postTitle }</div>
+				<div className="comment-list__header-date">{ formattedDate }</div>
+			</HeaderCake>
+		</div>
+	);
+};
+
+const mapStateToProps = ( state, { postId } ) => {
+	const site = getSelectedSite( state );
+	const siteId = getSelectedSiteId( state );
+	const siteSlug = getSelectedSiteSlug( state );
+
```

Comment:
Do we need these spaces? Can all the `const` declarations just be one block?

Reply:
While we normally cared about a single var declaration at the top of functions in es5 for variable hoisting, one def per line does tend to give us cleaner diffs. For example imagine if I removed the site const in a future PR.

Reply:
I think @kwight meant the empty lines between some groups of declarations.
At first there was a meaning behind that (started in the comment detail redesign), but it got quickly super messy, so there is honestly no need for them anymore.